tsnet/tsnet.go

```diff
@@ -88,6 +88,11 @@ type Server struct {
 	// If empty, the Tailscale default is used.
 	ControlURL string
 
+	// NotifyFunc is an optional callback function for engine updates from
+	// the LocalBackend. This can be used as an alternative to polling the
+	// LocalAPI in client applications.
+	NotifyFunc func(*ipn.Notify) (keepGoing bool)
```

Comment:
I think we need to make this more fine-grained for a stable API we can support long-term. This is kinda exposing some guts.

Also, WatchNotifications without letting the user specify what they want to watch isn't enough.

I'd rather put N different callbacks on here to let tsnet.Server users subscribe to specific changes and then we can map that to whatever the WatchNotification subscription bits are needed over time.

And then we can also shut down all the listeners on Close, and then not expose this keepGoing bool.